app/presentation/seed.py

```diff
@@ -122,17 +153,106 @@ def seed_db(search_client: SearchClient = Provide[Container.search_client]) -> N
         # Convertion du dataframe en string json séparée par des \n
         df_as_json = df.to_json(orient='records', lines=True)
 
-        click.echo("Seeding the database.")
-
         with click.progressbar(df_as_json.split('\n')) as bar:
             for json_document in bar:
                 if json_document != '':
                     # Convertion de la string json en dictionnaire
                     jdict = json.loads(json_document)
                     if jdict['resources_count'] != '0':
-                        search_client.index_dataset(Dataset(**jdict))
+                        dataset_service.feed(Dataset(**jdict))
+
+
+def process_reuse_catalog(reuse_service: ReuseService):
+    click.echo("Processing reuses data.")
+    with NamedTemporaryFile(delete=False) as reuse_fd:
+        download_catalog(current_app.config['REUSE_CATALOG_URL'], reuse_fd)
+    with NamedTemporaryFile(delete=False) as org_fd:
+        download_catalog(current_app.config['ORG_CATALOG_URL'], org_fd)
+    with open(reuse_fd.name) as reuse_csvfile, open(org_fd.name) as org_csvfile:
+        # Dataframe catalogue reuse
+        dfr = pd.read_csv(reuse_csvfile, dtype="str", sep=";")
+        # Dataframe catalogue orga
+        dfo = pd.read_csv(org_csvfile, dtype="str", sep=";")
+
+        # Récupèration de l'information "service public" depuis la colonne badge.
+        # Attribution de la valeur 4 si c'est un SP, 1 si ça ne l'est pas
+        dfo['orga_sp'] = dfo['badges'].apply(lambda x: 4 if 'public-service' in x else 1)
+
+        # Sauvegarde en mémoire du dataframe avec uniquement les infos pertinentes
+        dfo = dfo[['id', 'orga_sp', 'metric.followers']]
+
+        # Renommage de l'id de l'organisation et de la métrique followers
+        dfo = dfo.rename(columns={
+            'id': 'organization_id',
+            'metric.followers': 'orga_followers'
+        })
+
+        # Merge des deux dataframes (dataset et orga) pour n'en garder qu'un seul unique
+        # La colonne de jointure est l'id de l'organisation. On fait un merge de type left join
+        df = pd.merge(dfr, dfo, on='organization_id', how='left')
 
-        click.echo("Done.")
+        df['metric.views'] = df['metric.views'].astype(float)
+        df['metric.followers'] = df['metric.followers'].astype(float)
+        df['orga_followers'] = df['orga_followers'].astype(float)
+
+        mvbins = [-1, 0, 50, 500, 5000, df['metric.views'].max()]
+        df['reuse_views'] = pd.cut(df['metric.views'], mvbins, labels=list(range(1, 6)))
+        mfbins = [-1, 0, 2, 10, 40, df['metric.followers'].max()]
+        df['reuse_followers'] = pd.cut(df['metric.followers'], mfbins, labels=list(range(1, 6)))
+        fobins = [-1, 0, 10, 50, 100, df['orga_followers'].max()]
+        df['orga_followers'] = pd.cut(df['orga_followers'], fobins, labels=list(range(1, 6)))
```

Comment:
I'm not sure that we should keep the same normalization as the the one for datasets, but without quantitative data to test, we can't know what would be best.